src/units/udisplay.cpp

```diff
@@ -620,6 +621,13 @@ void unit_attack(display * disp, game_board & board,
 	assert(def.valid());
 	unit &defender = *def;
 	int def_hitpoints = defender.hitpoints();
+	const_attack_ptr weapon = attack.shared_from_this();
+	auto ctx = weapon->specials_context(attacker.shared_from_this(), defender.shared_from_this(), a, b, attacking, secondary_attack);
+	std::optional<decltype(ctx)> opp_ctx;
```

Comment:
What is `ctx`/`opp_ctx` created here?  They don't seem to be referenced anywhere else.

Reply:
i can't use weapon_leadership for what abilities used like weapon call [leading_anim] else the condition of activity don't be checked correctly.

Reply:
special_id_type_active don't detectspecial for animation without ctx/op_ctx